vertical-pod-autoscaler/enhancements/NNNN-max-allowed-recommendation/README.md

```diff
@@ -0,0 +1,45 @@
+# KEP-NNNN: Max allowed recommendation configurable
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+   - [Goals](#goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+   - [Test Plan](#test-plan)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+VPA currently allows configuring the maximum allowed recommendation per resource using the field `.spec.resourcePolicy.containerPolicies[].maxAllowed`. This helps ensure that recommendations don't exceed the allocatable capacity of the largest node. Without this limit, a high recommendation could make the Pod unschedulable, causing downtime. We aim to improve the VPA recommender by making it aware of the maximum node size, preventing it from generating recommendations that would render Pods unschedulable.
+
+## Motivation
+In dynamic environments, manual configuration of .maxAllowed can be error-prone, especially as cluster sizes and node configurations change over time. Automating this process by making the VPA recommender aware of the largest node size ensures more reliable scaling decisions. This prevents situations where Pods become unschedulable due to excessive resource recommendations.
+
+### Goals
+* Ensure Accurate and Schedulable Recommendations: Guarantee that the VPA recommender provides resource recommendations that do not exceed the allocatable capacity of the largest node, ensuring Pods remain schedulable and avoiding unnecessary downtime.
+* Support for Changing Environments: Adapt recommendations as the cluster size or node configurations evolve.
+
+## Proposal
+VPA could query the Cluster Autoscaler (CA) for feedback, potentially using the `provisioningrequest` API or a similar mechanism. This approach aligns well with CA’s design, as it is built to handle questions related to resource availability and scheduling. However, this solution comes with some challenges:  
+
+- Not all clusters use CA, and Karpenter currently does not support the `provisioningrequest` API and is unlikely to support it in the future.  
+- While we’ve previously discussed building a unified API supported by both CA and Karpenter, it is not a high priority and is unlikely to be implemented soon.  
+- Some clusters may not use any form of node autoscaling at all.  
+- Additionally, the `provisioningrequest` API, in its current form, is not optimized for this specific use case, meaning adjustments would be needed in CA to fully support it.  
```

Comment:
Is the proposal to use the ProvisioningRequest API from the cluster-autoscaler?
Based on the language used here, it sounds like the plan is to not use it?

Reply:
The proposal is to use the Cluster Autoscaler (CA) in the cluster through the ProvisioningRequest API, but there are some downsides mentioned in this KEP. For example, the cluster might be using Karpenter instead of CA.

Reply:
I’ve outlined the downsides we need to consider if we decide to move forward with this approach.